src/components/certificate-table/CertificatesTable.tsx

```diff
@@ -2,48 +2,97 @@ import './certificate-table.css';
-  { Header: 'Valid from', accessor: 'validFrom' },
-  { Header: 'Valid to', accessor: 'validTo' },
-];
-
-/**
- * Example - content wrapper for the certificates table
- * Description - Component which has the certificates table
- * and create certificate button
- * @returns JSX Element
- */
-
-function CertificatesTable(): JSX.Element {
+interface Certificate extends BaseCertificate {
+  actions: React.ReactNode;
+}
+
+const CertificatesTable: React.FC = () => {
   const { certificates } = useCertificates();
+  const navigate = useNavigate();
+  const [dropdownVisible, setDropdownVisible] = useState<number | null>(null);
+  const dropdownRef = useRef<HTMLDivElement | null>(null)
+
+  const handleEdit = (id: number) => {
+    navigate(`/certificates/edit/${id}`);
+  };
+
+  const handleDropdownToggle = (index: number) => {
+    setDropdownVisible(dropdownVisible === index ? null : index);
+  };
+
+  const handleClickOutside = (event: MouseEvent) => {
+    if (dropdownRef.current && !dropdownRef.current.contains(event.target as Node)) {
+      setDropdownVisible(null);
+    }
+  };
+
+  useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  const columns: Column<Certificate>[] = [
+    {
+      Header: ' ',
+      accessor: 'actions',
+      render: (row, rowIndex) => (
+        <div
+          className="cog-container"
+          onClick={() => handleDropdownToggle(rowIndex)}
+          ref={ dropdownRef}
+        >
+          <CogIcon />
+          {dropdownVisible === rowIndex && (
+            <div className="dropdown-menu">
+              <button onClick={() => handleEdit(row.id)}>Edit</button>
+              <button onClick={() => console.log('delete clicked')}>
+                Delete
+              </button>
+            </div>
+          )}
+        </div>
+      ),
+    },
+    { Header: 'Supplier', accessor: 'supplier' },
+    { Header: 'Certificate type', accessor: 'certificateType' },
+    { Header: 'Valid from', accessor: 'validFrom' },
+    { Header: 'Valid to', accessor: 'validTo' },
+  ];
+
+  const dataWithActions = certificates.map((cert) => ({
+    ...cert,
+    actions: null,
+  }));
 
   return (
     <section className="certificates-table" aria-labelledby="certificatesTitle">
       <h1>Certificates</h1>
       <span className="new-certificate">
         <Button variation="contained" size="medium">
           <Link
-            to="/certificates/new"
+            to="/new-certificate"
             style={{ color: 'inherit', textDecoration: 'none' }}
           >
             Add New Certificate
           </Link>
         </Button>
       </span>
-
-      <Table<Certificate> columns={columns} data={certificates} />
+      <Table<Certificate> columns={columns} data={dataWithActions} />
     </section>
   );
-}
+};
 
 export default CertificatesTable;
```

Comment:
### Recommendations:
- **Separate Component for Action Menu:**
The action menu (cog icon with dropdown) should be extracted into its own component for better reusability and maintainability.
```TSX
const ActionMenu = ({ row, onEdit, onDelete }) => {
  const [isOpen, setIsOpen] = useState(false);
  const dropdownRef = useRef(null);

  useClickOutside(dropdownRef, () => setIsOpen(false));

  return (
    <div className="cog-container" onClick={() => setIsOpen(!isOpen)} ref={dropdownRef}>
      <CogIcon />
      {isOpen && (
        <div className="dropdown-menu">
          <button onClick={() => onEdit(row.id)}>Edit</button>
          <button onClick={() => onDelete(row.id)}>Delete</button>
        </div>
      )}
    </div>
  );
};
```
- **Optimize Rendering**:
Memoize the columns array to prevent unnecessary re-renders.
```TSX
const columns = useMemo(() => [
  // ... column definitions
], [handleEdit, handleDelete]);
```
- **Naming Convention**:
Consider renaming `dataWithActions` to something more descriptive, as it's not actually adding actions but a null property.